build_system/seqan3-config.cmake

```diff
@@ -359,6 +359,7 @@ else ()
             set (C++17FS_LIB ${_LIB})
             break ()
         endif ()
+        set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES_ORIGINAL})
```

Comment:
Here we restore `CMAKE_REQUIRED_LIBRARIES ` to not include the last, failed `${_LIB}` and wich is used only during the compilation with `check_include_file_cxx` and `check_include_file_cxx`.

Reply:
But the variable is reset in every loop anyway. The only time it is not reset is when the loop is exited via break (the correct flag is found) or if not correct flag is found in which case the whole thing fails anyway. Or am I missing something?

Reply:
Well, we don't need this change here really. 
But it make the code more robust. The problem was that in posterior, unrelated tests I was having an strange behavior, see lines [range3 test](https://github.com/qPCR4vir/seqan3/commit/bfeecbf4eb3b0d39e96fe9b47c80775d3d25cde2#diff-81042d2e00dba45e1e36277ce17139d3R391) and [error](https://github.com/qPCR4vir/seqan3/commit/bfeecbf4eb3b0d39e96fe9b47c80775d3d25cde2#diff-81042d2e00dba45e1e36277ce17139d3R404). Debbugging cmake is not ... easy. It was wild, because it report it find the modules, but then failed to uses it. It silently report old, cached results, no matter what preconditions you change: you need to invalidate the cache or unset the test control variable (I did). In fact I changed my local copy of `check_include_file_cxx ` to report what it was really doing. I mean, first, the tests failed, but when I detected the problem with the flag this other test still "failed"... but the real seqan3 gtests were susseful.
In this position this reset is more local and easy to follow - and it is executed very few times. In fact, here- not all if the flag works.